convert.py

```diff
@@ -15,65 +15,114 @@
 
 
-def convert(dirname):
+def parse_args():
+    # Read arguments: path to model.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--fluid_model", required=True, help="Input PaddlePaddle Fluid model.")
+    parser.add_argument(
+        "--onnx_model_dir",
+        required=False,
+        help="The output directory for ONNX model.")
+    args = parser.parse_args()
+    return args
+
+
+def convert(args):
     # Read the model files.
     place = fluid.CPUPlace()
     exe = fluid.Executor(place)
 
     inference_scope = fluid.core.Scope()
     with fluid.scope_guard(inference_scope):
         [inference_program, feed_target_names,
-         fetch_targets] = fluid.io.load_inference_model(dirname, exe)
+         fetch_targets] = fluid.io.load_inference_model(args.fluid_model, exe)
 
         # Using blocks in programs, create nodes using:
         onnx_nodes = []
-        all_inputs = []
-        for block in inference_program.blocks:
-            all_inputs += [
-                paddle_variable_to_onnx_tensor(v, block) for v in block.vars
-                if v not in ['feed', 'fetch']
-            ]
 
+        # Load parameters
+        global_block = inference_program.global_block()
+        for var_name in global_block.vars:
+            var = global_block.var(var_name)
+            if var_name not in ['feed', 'fetch'] and var.persistable:
+                param = fluid.executor.fetch_var(var_name, inference_scope)
+                param_node = helper.make_node(
+                    'Constant',
+                    inputs=[],
+                    outputs=[var_name],
+                    value=helper.make_tensor(
+                        name=var_name,
+                        dims=var.shape,
+                        data_type=PADDLE_TO_ONNX_DTYPE[var.dtype],
+                        vals=param.flatten().tolist()))
+                onnx_nodes.append(param_node)
+
+        # Create inputs
+        inputs = [
+            paddle_variable_to_onnx_tensor(v, global_block)
+            for v in feed_target_names
+        ]
+
+        # Create outputs
+        fetch_target_names = [
+            fetch_target.name for fetch_target in fetch_targets
+        ]
+        outputs = [
+            paddle_variable_to_onnx_tensor(v, global_block)
+            for v in fetch_target_names
+        ]
+
+        # Create nodes
+        for block in inference_program.blocks:
             for op in block.ops:
-                if op.type in ops.PADDLE_TO_ONNX:
-                    # TODO(varunarora): Attributes.
-                    # TODO(varunarora): Use the modifier function to make the
-                    # transformation.
-                    node_proto = helper.make_node(
-                        ops.PADDLE_TO_ONNX[op.type][0], op.input_arg_names,
-                        op.output_arg_names)
+                if op.type in ops.node_maker:
+                    # TODO(kuke): deal with the corner case that vars in 
```

Comment:
Aren't unique names generated, no matter how local the scope is? You probably know more here..

Reply:
Yes, seems that duplicated names are allowed for local variables. I am discussing with others working on framework development, to figure out a proper solution. So if you have any idea, please feel free to speak out.

Reply:
No ideas here, you can leave the comment as is, but I don't think you should worry about a conflict here

Reply:
Yes, I believe that we can figure a way to solve this conflicts.